### PR TITLE
fix bug that misidentified winner when a play was anonymous

### PR DESCRIPTION
**Pull request by iankhou**
# Description

Previously, backend logic defaulted to `null` when an anonymous player forfeited.

Now, the `winnerId` is set to `"anonymous"` when an anonymous player wins via game completion or forfeit.

## Type of Change

- [ ] New feature
- [X] Bug fix
- [ ] Breaking change
- [ ] Documentation update

## Checklist

- [ ] Unit/integration tests
- [ ] Documentation

### CODE REVIEW

File: src/lzqgame.ts [hunk 1]
```diff
@@ -258,26 +258,28 @@ async function playerInitialBoard(
...
+                if (!myGame) {
+                    console.error('Game not found.');
+                    this.emit('error', [`$Game not found: ${room}`]);
+                    return;
+                }
 
-            const playerIndex = myGame.players.indexOf(instPlayerName);
-            console.info(`playerIndex: ${playerIndex}`);
+                const playerIndex = myGame.players.indexOf(instPlayerName);
+                console.info(`playerIndex: ${playerIndex}`);
```
Comment (fireteam99) [inline src/lzqgame.ts:274]: some weird formatting happening here?

File: src/lzqgame.ts [hunk 2]
```diff
@@ -477,8 +479,8 @@ async function playerMakeMove(
                 moves: [...moveHistory, pendingMove],
             });
             myGame = await getGame(room);
-            
-          if (!myGame) {
+
+            if (!myGame) {
```
Comment (fireteam99) [inline src/lzqgame.ts:483]: more weird formatting

**Review by fireteam99 — COMMENTED** [review]
something seems to be up with the formatter but otherwise looks fine

File: src/lzqgame.ts (hunk 2 above)
Comment (iankhou) [reply src/lzqgame.ts:483]: It was previously too far left by one tab, I think? Auto formatter fixed this.

File: src/lzqgame.ts (hunk 1 above)
Comment (iankhou) [reply src/lzqgame.ts:274]: Look at the file: https://github.com/chinese-board-games/luzhanqi-backend/blob/4ac44a11d0e151e30bd8bb772e5d700ddc11651a/src/lzqgame.ts

**Pull request merged**